main/config.c

```diff
@@ -362,17 +372,52 @@ struct ast_config_include *ast_include_new(struct ast_config *conf, const char *
 	else
 		inc->included_file = ast_strdup(included_file);
 
-	inc->exec = is_exec;
-	if (is_exec)
-		inc->exec_file = ast_strdup(exec_file);
-
 	if (!inc->include_location_file
 		|| !inc->included_file
-		|| (is_exec && !inc->exec_file)) {
+		|| ((include_type > 0) && !inc->exec_file)) {
 		ast_includes_destroy(inc);
 		return NULL;
 	}
 
+	inc->include_type = include_type;
+
+	make_fn(fn, sizeof(fn), inc->included_file, from_file);
+	if (inc->include_type == 0) {
+		/* if #include */
+		inc_file = ast_calloc(1, sizeof(*inc_file));
```

Comment:
Since this struct just stores a string, I would change it to:

struct ast_config_included_file {
   const char *filename; // fn could be function or filename, so use the full world
  AST_LIST_ENTRY(ast_config_included_file) next;
  char data[];
}

Then, do ast_calloc(1, sizeof(*inc_file) + strlen(fn) + 1); Then you can just strcpy fn to data and set the filename pointer to data. It's a common trick in this codebase and others that will save an allocation and reduce memory fragmentation.
Of course, if you can explicitly initialize next to NULL then you can also use malloc instead of calloc